spec/core/data_structures.md

```diff
@@ -4,195 +4,243 @@ Here we describe the data structures in the Tendermint blockchain and the rules
+| ValidatorAddress | [Address](#address)         | Address of the validator                                                                                      | Must be of length 20                                              |
+| Timestamp        | [Time](#time)               | This field will vary from `CommitSig` to `CommitSig`. It represents the timestamp of the validator.               | [Time](#time)                                                     |
+| Signature        | [Signature](#signature)     | Signature corresponding to the validators participation in consensus.                                         | The length of the signature must be > 0 and < than  64            |
 
-const (
- // BlockIDFlagAbsent - no vote was received from a validator.
- BlockIDFlagAbsent BlockIDFlag = 0x01
- // BlockIDFlagCommit - voted for the Commit.BlockID.
- BlockIDFlagCommit = 0x02
- // BlockIDFlagNil - voted for nil.
- BlockIDFlagNil = 0x03
-)
+NOTE: `ValidatorAddress` and `Timestamp` fields may be removed in the future
+(see [ADR-25](https://github.com/tendermint/tendermint/blob/master/docs/architecture/adr-025-commit.md)).
+
+## BlockIDFlag
 
-type CommitSig struct {
- BlockIDFlag      BlockIDFlag
- ValidatorAddress Address
- Timestamp        time.Time
- Signature        []byte
+BlockIDFlag represents which BlockID the [signature](#commitsig) is for.
+
+```go
+enum BlockIDFlag {
+  BLOCK_ID_FLAG_UNKNOWN = 0;
+  BLOCK_ID_FLAG_ABSENT  = 1;
+  BLOCK_ID_FLAG_COMMIT  = 2;
+  BLOCK_ID_FLAG_NIL     = 3;
 }
 ```
 
-NOTE: `ValidatorAddress` and `Timestamp` fields may be removed in the future
-(see
-[ADR-25](https://github.com/tendermint/tendermint/blob/master/docs/architecture/adr-025-commit.md)).
-
 ## Vote
 
 A vote is a signed message from a validator for a particular block.
-The vote includes information about the validator signing it.
+The vote includes information about the validator signing it. When stored in the blockchain or propagated over the network, votes are encoded in Protobuf.
+
+| Name             | Type                            | Description                                                                                 | Validation                                                                                           |
+|------------------|---------------------------------|---------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------|
+| Type             | [SignedMsgType](#signedmsgtype) | [SignedMsgType](#signedmsgtype)                                                             | A Vote is valid if its corresponding fields are included in the enum [signedMsgType](#signedmsgtype) |
```

Comment:
I would suggest changing the description to:

```
Either prevote or precommit
```
and removing the bottom part ... "There are two types of votes"

Reply:
Also do we want to add the data struct of the Proposal

Reply:
Yes, there are few things missing. I didn't want bloat this PR, is it okay if I do a followup with more changes?